webrender/src/renderer.rs

```diff
@@ -1722,34 +1723,78 @@ impl Renderer {
             self.gpu_data_textures[self.gdt_index].init_frame(&mut self.device, frame);
             self.gdt_index = (self.gdt_index + 1) % GPU_DATA_TEXTURE_POOL;
 
-            let mut src_id = self.dummy_cache_texture_id;
-
-            for (pass_index, pass) in frame.passes.iter().enumerate() {
-                let (do_clear, size, target_id) = if pass.is_framebuffer {
-                    (self.clear_framebuffer || needs_clear,
-                     framebuffer_size,
-                     None)
+            let mut src_color_id = self.dummy_cache_texture_id;
+            let mut src_alpha_id = self.dummy_cache_texture_id;
+
+            for pass in &mut frame.passes {
+                let size;
+                let clear_color;
+                let projection;
+
+                if pass.is_framebuffer {
+                    clear_color = if self.clear_framebuffer || needs_clear {
+                        Some(frame.background_color.map_or(self.clear_color.to_array(), |color| {
+                            color.to_array()
+                        }))
+                    } else {
+                        None
+                    };
+                    size = framebuffer_size;
+                    projection = Matrix4D::ortho(0.0,
+                                                 size.width as f32,
+                                                 size.height as f32,
+                                                 0.0,
+                                                 ORTHO_NEAR_PLANE,
+                                                 ORTHO_FAR_PLANE)
                 } else {
-                    (true, &frame.cache_size, Some(self.render_targets[pass_index]))
-                };
+                    size = &frame.cache_size;
+                    clear_color = Some([1.0, 1.0, 1.0, 0.0]);
+                    projection = Matrix4D::ortho(0.0,
+                                                 size.width as f32,
+                                                 0.0,
+                                                 size.height as f32,
+                                                 ORTHO_NEAR_PLANE,
+                                                 ORTHO_FAR_PLANE);
+                }
 
-                for (target_index, target) in pass.targets.iter().enumerate() {
-                    let render_target = target_id.map(|texture_id| {
+                self.device.bind_texture(TextureSampler::CacheA8, src_alpha_id);
```

Comment:
There is an option to call them `CacheAlpha`/`CacheColor`, but I'm not seeing it particularly superior